include/veriblock/pop/mempool.hpp

```diff
@@ -378,6 +378,23 @@ struct MemPool {
   payload_value_sorted_map<T>& getInFlightMapMut() {
     return const_cast<payload_value_sorted_map<T>&>(this->getInFlightMap<T>());
   }
+
+  struct EndorsedAltComparator {
+    EndorsedAltComparator(const MemPool& parent) : parent_(parent) {}
+    int operator()(const std::shared_ptr<ATV>& a,
+                   const std::shared_ptr<ATV>& b) const;
+
+   private:
+    const MemPool& parent_;
+  };
+
+  struct TxFeeComparator {
```

Comment:
I thought we could have 1 comparator, which does comparison by TX FEE first, then if it is equal will continue with other criterias. How does multiple comparators align with this idea?

Reply:
I prefer to do it in a functional way and construct combined comparator from separate comparators. Unfortunately C++ does not have something like std::combine, so no advantage here but still looks a bit better to me.